### DISCUSSION TITLE
RTMFP with Mona server

### DISCUSSION THREAD

**Discussion by raresn** [airsdk/Adobe-Runtime-Support#1938] [Q&A]
We are trying to stop doing pulling of data from a server every x seconds, but insetead commnunicate with a Mona server using RTMFP that tells all users in a group that there is an update and that they should pull.
First of all, is there another (better) way of doing this?

We are doing netGroups to post messaages and broadcast to all users connected in the same group, but accodring to: https://helpx.adobe.com/adobe-media-server/dev/rtmfp-groups.html#bootstrap_a_peer_to_a_group
"If two sets of clients use the same GroupSpecifier, and never touch, they are in separate groups. If the two groups touch, they merge into a single larger group."

Our code is this:

```
myGrpSpec = new GroupSpecifier(groupName);
myGrpSpec.postingEnabled = true;
myGrpSpec.ipMulticastMemberUpdatesEnabled = true;
myGrpSpec.serverChannelEnabled = true;
myGrpSpec.objectReplicationEnabled = true;
ng = new NetGroup(nc, myGrpSpec.groupspecWithAuthorizations());
```

If connection is successful, then we call:
`ng.post({text: text, seed: Math.random()});`

This worked around 6 months ago, and now we want to reimplement this, in a better way, and it doesn't anymore.
It didn't work on 3G/4G connection in the past either. just WiFi.

Does anyone have experience with real time communication? Any tips on how we should handle this?

**Comment by Dextercitox** [comment]
Hi !!!

We're using actually a RTMFP / FIrebase combo to achieve a real time communication with offline support. Devices use both and receive duplicated events ( we just clean them internally ) and if you lost internet connection, RTMFP is still working in the local WiFi. 

Firebase is actually a great option for real time communication and the Distriqt extension works very well :
https://docs.airnativeextensions.com/docs/firebase/

With that you have the real time events all the time, and it's up to you how to implement the groups, permissions and so on.

Hope that helps !

**Comment by raresn** [comment]
@Dextercitox, for RTMFP, you use MONA? or what other system?

Using firebase - you are reffering to push notifications?(If so i would very much like to avoid that, because not all users allow the push notifications.)

**Comment by Dextercitox** [comment]
Hi @raresn  !!

For RTMFP we create a local internal network directly from AS3, because for us RTMFP is used only in local network ( it's like a backup network ).

In the case of Firebase, we dont use push notifications. We use Real Time Database ( you have also FireStore available ) and the strong point is that you can have a listener attached to any node of the DB ( is a noSQL DB, like a big JSON ) and any of your users will be notified of any change in the data. Those changes are real time, actually we receive the Firebase Events faster than the RTMFP ones ( and those are local !! ) so it's quite easy to have peer to peer communication.

**Reply by raresn** [reply]
I did not think about the real time database!
Will definitely give it a shot!

**Reply by raresn** [reply]
What about pricing? 
Do you find it acceptable? We will most likely need more than 100 coonections at a time.
1 account had around 4-5 users, and we got around 200 accounts. If all of them get connected we will most likely go over the 100 simultanous connections.

**Reply by Dextercitox** [reply]
Actually very cheap, we usually handle more than 300 simultaneous connections and about 40Gb of data every month and pay about 32€/month .... No headache about maintaining servers, infrastructure, backups ...

**Reply by raresn** [reply]
Another question: any reason why you didn't use RTMFP thoughout on all platforms?

**Reply by Dextercitox** [reply]
Because we also make the devices work "online", they have RTMFP only for the communication in the internal network. :)

**Reply by Dextercitox** [reply]
Because we also make the devices work "online", they have RTMFP only for the communication in the internal network. :)
